base/src/main/java/io/spine/type/Type.java

```diff
@@ -42,6 +44,20 @@
 @Internal
 public abstract class Type<T extends GenericDescriptor, P extends Message> {
 
+    /**
+     * A cache of known classes per {@linkplain #javaClassName() Java class name}.
+     *
+     * <p>Saves the memory by limiting the maximum number of cached objects.
+     *
+     * @implNote {@link com.google.common.cache.LoadingCache LoadingCache} implementation
+     *         would make dealing with the exceptions a bit more difficult to maintain and read.
+     *         Which is why a straightforward get-put approach with a plain {@code Cache}
+     *         implementation is used.
+     */
+    private static final Cache<String, Class<?>> knownClasses = CacheBuilder.newBuilder()
+                                                                            .maximumSize(1_000)
+                                                                            .build();
```

Comment:
Please use a loader instead of manually creating instances in `javaClass()`.

Also, I would consider making the `1_000` a constant, or better yet, a configurable property, since different environments may require different cache sizes.